Adapt the width of the pricing fields

nathanss opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Changes the width of the pricing fields based on the screen's width.

I'm not sure how we want to determine the width of the fields. @jarekmorawski mentioned the 1032px width rule on issue #35180, so that's what I adopted in the first place.

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #35538.

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested (including pre-conditions, configuration, steps to take and expected results). It may help to write your instructions using pseudocode -- as if you're telling a computer how to execute the test. -->

1. Go to the new Add Product page (/wp-admin/admin.php?page=wc-admin&path=%2Fadd-product)
2. Check if the width of the pricing fields are 50% when the screen width is larger than 1032px
3. Adjust the screen width to a smaller value and check if the width of the pricing fields turn into 100%

<!-- End testing instructions -->

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

nathanss committed:
Add css class to change the input width based on the screen's width

nathanss committed:
Changelog

mattsherman commented on plugins/woocommerce-admin/client/products/sections/pricing-section.scss:
```diff
@@ -4,3 +4,10 @@
 	}
 }
+
+.woocommerce-field-width {
```
This class name feels too generic (I recognize we have other generic class names in this file... we should clean those up too in a future PR). Since it is in the pricing section, we should name it something like:

`product-pricing-section__price-field`

Or, move it somewhere more general, like https://github.com/woocommerce/woocommerce/blob/736593ef1526ac1aa5495851789690b26eb2d657/plugins/woocommerce-admin/client/products/product-page.scss and call it something like:

```
.woocommerce-add-product,
.woocommerce-edit-product {
    .half-width-field {
        // props go here
    }
}
```

I'm not sold on the `half-width-field`... part of me likes it and part of me thinks we should list our the classname of every specific type of field that is going to use this instead. What do you think?

nathanss replied:
I'd say we keep like `half-width-field` specially since it seems we will be adding this to more fields in the future. We can change in the future if needed

mattsherman commented on plugins/woocommerce-admin/client/products/sections/pricing-section.scss:
```diff
@@ -4,3 +4,10 @@
 }
+
+.woocommerce-field-width {
+	@media only screen and (max-width: 1032px) {
```
We should use the following instead:

```
@include breakpoint( '<960px' ) {
    width: 100%;
}
```

Using the `breakpoint` mixin from https://github.com/woocommerce/woocommerce/blob/112b9ac67a12a86e33d1d9d47027c76f17522083/packages/js/internal-style-build/abstracts/_breakpoints.scss

Since that mixin doesn't support the `1032px` breakpoint, we should check with @jarekmorawski to see what breakpoint we should use (`960px` feels right to me, and is the breakpoint where WP collapses the admin sidebar).

mattsherman commented on plugins/woocommerce-admin/client/products/sections/pricing-section.tsx:
```diff
@@ -128,6 +128,7 @@ export const PricingSection: React.FC = () => {
 					>
 						<InputControl
 							{ ...regularPriceProps }
+							className="woocommerce-field-width"
```
I'd put this in `currencyInputProps` above. That way, it will get merged I believe, and not just override any other classes already set on the component via `regularPriceProps` or anything else up the chain.

nathanss replied:
Adding it to `currencyInputProps` is not yielding the expected result. It seems to be added to multiple elements, resulting in a 25% width

<img width="733" alt="image" src="https://user-images.githubusercontent.com/13437655/201669369-42021a50-8d5e-497b-bfb0-293a0dfe2cfa.png">

I think we need a way of adding it only to the `InputControl`. Do you know a better way?

mattsherman replied:
Ah, wait... we are wrapping the `InputControl` in a `BaseControl`, and we are pulling out `className` to set on the `BaseControl`. So, I'd still suggest putting it in `currencyInputProps`, but only set it on either `BaseControl` or `InputControl`... not sure which one would be best, probably `InputControl`, since the `label` or `help` could be longer (as your screenshot shows).

Honestly, with what we are doing here with `BaseControl` and `InputControl`, we ideally should have a new component such as `CurrencyInput` that takes care of all of this. Otherwise, things are quickly going to get messy and difficult to maintain.

mattsherman commented on plugins/woocommerce-admin/client/products/sections/pricing-section.tsx:
```diff
@@ -149,6 +150,7 @@ export const PricingSection: React.FC = () => {
 					>
 						<InputControl
 							{ ...salePriceProps }
+							className="woocommerce-field-width"
```
Same as above.

mattsherman commented on plugins/woocommerce/changelog/update-pricing-inputs-width:
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: enhancement
+
+Adapt the width of pricing fields in new product management experience
```
I'd probably make this comment a little more specific. "Adapt" feels a bit vague.

mattsherman reviewed (changes requested):
Nice start on this @nathanss! I left a few comments/pointers to things you should consider to make this implementation a bit more robust. Mainly from a naming standpoint and using our standard `breakpoint` mixin and a standard breakpoint value.

mattsherman commented:
@nathanss I notice the "Run lint checks" was canceled. Not really sure (sometimes that just happens, I think). You can re-run checks by clicking "Details" on the failing check and click the "Re-run jobs" button. Of course, pushing new commits will also trigger a re-run, so you don't really need to worry about this until you are at the state where you are looking for a "final" review!

nathanss committed:
Move class from pricing-section.scss to product-page.scss and use breakpoint 960px mixin

nathanss committed:
Move className from currencyInputProps to InputControl

nathanss committed:
Improve changelog

nathanss committed:
Move 'half-width-field' class to currencyInputProps instead of adding it manually for the two currency fields

nathanss committed:
Add 'half-width-field' class to SKU and Shipping Class

Sent as a parameter to getInputProps and getSelectControlProps to avoid overwriting any additional className

mattsherman commented on plugins/woocommerce/changelog/update-pricing-inputs-width:
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: enhancement
+
+Change the width of pricing fields to 50% in big screens (>960px) in new product management experience
```
This should be updated since it is not just the pricing fields that are being updated.

mattsherman commented on plugins/woocommerce-admin/client/products/sections/pricing-section.tsx:
```diff
@@ -123,7 +124,6 @@ export const PricingSection: React.FC = () => {
 				<CardBody>
 					<BaseControl
 						id="product_pricing_regular_price"
-						className={ regularPriceProps?.className ?? '' }
```
I'm okay with this change. As mentioned before, ideally we should abstract away this `<BaseControl>` usage so we aren't directly using it here. That is for the future though.

mattsherman reviewed (changes requested):
Just a minor changelog update.

Other than that, looks good and tests well.

nathanss committed:
Update changelog

mattsherman reviewed (approved):
Thanks for updating the changelog. Looks good. Nice work!
